HumanSpaceProgram/Assets/HSP.ReferenceFrames/ReferenceFrames/OrientedNonInertialReferenceFrame.cs

```diff
@@ -0,0 +1,217 @@
+        {
+            return _inverseRotation * Vector3Dbl.Subtract( globalPosition, _position );
+        }
+
+
+        public Vector3 TransformDirection( Vector3 localDirection )
+        {
+            return (Vector3)(_rotation * localDirection);
+        }
+        public Vector3 InverseTransformDirection( Vector3 absoluteDirection )
+        {
+            return (Vector3)(_inverseRotation * absoluteDirection);
+        }
+
+
+        public QuaternionDbl TransformRotation( QuaternionDbl localRotation )
+        {
+            return _rotation * localRotation;
+        }
+        public QuaternionDbl InverseTransformRotation( QuaternionDbl airfRotation )
+        {
+            return _inverseRotation * airfRotation;
+        }
+
+
+        public Vector3Dbl TransformVelocity( Vector3Dbl localVelocity )
+        {
+            return Vector3Dbl.Add( _rotation * localVelocity, _velocity );
+        }
+        public Vector3Dbl InverseTransformVelocity( Vector3Dbl globalVelocity )
+        {
+            return _inverseRotation * Vector3Dbl.Subtract( globalVelocity, _velocity );
+        }
+
+
+        public Vector3Dbl TransformAcceleration( Vector3Dbl localAcceleration )
+        {
+            return Vector3Dbl.Add( _rotation * localAcceleration, _acceleration );
+        }
+
+        public Vector3Dbl InverseTransformAcceleration( Vector3Dbl globalAcceleration )
+        {
+            return _inverseRotation * Vector3Dbl.Subtract( globalAcceleration, _acceleration );
+        }
+
+
+        public Vector3Dbl TransformAngularVelocity( Vector3Dbl localAngularVelocity )
+        {
+            return Vector3Dbl.Subtract( _rotation * localAngularVelocity, _angularVelocity );
+        }
+        public Vector3Dbl InverseTransformAngularVelocity( Vector3Dbl globalAngularVelocity )
+        {
+            return _inverseRotation * Vector3Dbl.Subtract( globalAngularVelocity, _angularVelocity );
+        }
+
+
+        public Vector3Dbl TransformAngularAcceleration( Vector3Dbl localAngularAcceleration )
+        {
+            return Vector3Dbl.Add( _rotation * localAngularAcceleration, _angularAcceleration );
+        }
+        public Vector3Dbl InverseTransformAngularAcceleration( Vector3Dbl globalAngularAcceleration )
+        {
+            return _inverseRotation * Vector3Dbl.Subtract( globalAngularAcceleration, _angularAcceleration );
+        }
+
+
+        public Vector3Dbl GetTangentialVelocity( Vector3Dbl localPosition )
+        {
+            // Since the output is in absolute space, the orientation matters here.
+            localPosition = _rotation * localPosition;
+            return Vector3Dbl.Cross( AngularVelocity, localPosition );
+        }
+
+        public Vector3Dbl GetFicticiousAcceleration( Vector3Dbl localPosition, Vector3Dbl localVelocity )
+        {
+            // Everything is in local (including the returned values), so the orientation is irrelevant.
+
+            // TODO - handle near-zeroes in the terms.
```

Comment:
Don't forget about todo

Reply:
After analyzing it, it appears that the near-zero terms should be handled fine (and 0 vector results in the accelerations being 0, which is correct a well), not sure why I wrote that.

I even tested it in game by setting the current scene frame to a non-inertial one.